WordPress/gutenberg
# Global Syles: Apply fallback background color to typography elements

Fixes #59338

## What?

This PR applies a fallback background color to icons and previews in the Global Styles Elements panel.



## Why?

The current logic uses the background color of each element. For example, if it is a link element, `elements.link.color.background` is referenced. However, if the text is close to white and does not have an explicit background color, it will be difficult to see the text.

## How?

Use `color.background` as a fallback background color in case no explicit background color is defined for that element. This behavior should actually be close to what you see visually in the frontend or editor canvas.

## Testing Instructions

- Open the Site Editor and access Global Styles > Typography.
- All elements should now have a background color applied to them.
- When you click on any element, the background color should also be applied in the preview.
- Repeat the above steps for several styles.

## Screenshots or screencast 

### Icon

| Style | Before | After |
|--------|--------|--------|
| Default | ![image](https://github.com/WordPress/gutenberg/assets/54422211/47590285-e726-4db6-8c18-a060d6200673)| ![image](https://github.com/WordPress/gutenberg/assets/54422211/b1923621-21dd-42f5-8fde-7ee1ed65cf3d) |
| Maelstrom | ![image](https://github.com/WordPress/gutenberg/assets/54422211/47e1ea02-3b79-4d64-9c00-6f684f1e2b60) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/641b778f-9297-4985-9ea9-8de2fb5d49d5) |
| Onyx | ![image](https://github.com/WordPress/gutenberg/assets/54422211/d848da9a-c22d-4bc1-8c23-146887918396) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/3042f1cb-8ba5-4b6a-be6f-98a15e622d3b) | 

### Preview

| Style | Before | After |
|--------|--------|--------|
| Default | ![image](https://github.com/WordPress/gutenberg/assets/54422211/418bebba-b9c9-4bb5-ab42-61aeaf816f4a)| ![image](https://github.com/WordPress/gutenberg/assets/54422211/9c2a18f3-8abf-4ccc-ad87-130cfdf6ea83) |
| Maelstrom | ![image](https://github.com/WordPress/gutenberg/assets/54422211/d267a16c-5dd0-401c-bf05-0201e21b306d) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/90e79d2a-f957-4f05-851a-e9c885c983bf) |
| Onyx | ![image](https://github.com/WordPress/gutenberg/assets/54422211/0b76ffed-e4b9-469e-a872-1cdc6d47fee6) | ![image](https://github.com/WordPress/gutenberg/assets/54422211/2f936eff-31c9-4d14-8b8f-bd91da86d66d) |
